android/src/play/java/com/dooboolab/RNIap/RNIapModule.java

```diff
@@ -676,4 +675,77 @@ private void sendEvent(
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchaseHistoryAsync( BillingClient.SkuType.SUBS, (billingResult, list) -> {
+        WritableMap item = Arguments.createMap();
+        for (int i = 0; i < list.size(); i++) {
+          PurchaseHistoryRecord purchase = list.get(i);
+          if (purchase.getSkus().get(0).equalsIgnoreCase(sku)) {
+            promise.resolve(true);
+            return;
+          }
+        }
+        promise.resolve(false);
+      });
+    });
+  }
+
+
+  @ReactMethod
+  public void getSubscriptionTransactionDetails(final String sku, final Promise promise) {
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchasesAsync( BillingClient.SkuType.SUBS, (billingResult, list) -> {
+        WritableMap item = Arguments.createMap();
+        Purchase purchase = list.get(0);
+        if (purchase.getSkus().get(0).equalsIgnoreCase(sku)) {
+          item.putString("productId", purchase.getSkus().get(0));
+          item.putString("orderId", purchase.getOrderId());
+          item.putString("purchaseToken", purchase.getPurchaseToken());
+          item.putString("purchaseTime", String.valueOf(purchase.getPurchaseTime()));
+          item.putString("transactionReceipt", purchase.getOriginalJson());
+          item.putString("purchaseToken", purchase.getPurchaseToken());
+          item.putString("dataAndroid", purchase.getOriginalJson());
+          item.putString("signatureAndroid", purchase.getSignature());
+        }
+        promise.resolve(item);
+
+      });
+
+    });
+
+  }
+
+  @ReactMethod
+  public void listOwnedProducts(final String type, final Promise promise) {
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchasesAsync( 
+        type, 
+        (billingResult, list) -> {
+          List<Purchase> purchases = list;
+          WritableArray items = Arguments.createArray();
+
+          for(int i = 0; i < purchases.size(); i ++) {
+            Purchase purchase = purchases.get(i);
+            WritableMap item = Arguments.createMap();
+            item.putString("productId", purchase.getSkus().get(0));
+            item.putString("orderId", purchase.getOrderId());
+            item.putString("purchaseToken", purchase.getPurchaseToken());
+            item.putString("purchaseTime", String.valueOf(purchase.getPurchaseTime()));
+            item.putString("transactionReceipt", purchase.getOriginalJson());
+            item.putString("purchaseToken", purchase.getPurchaseToken());
+            item.putString("dataAndroid", purchase.getOriginalJson());
+            item.putString("signatureAndroid", purchase.getSignature());
+            if (type.equals(BillingClient.SkuType.SUBS)) {
+              item.putBoolean("autoRenewingAndroid", purchase.isAutoRenewing());
+            }
+            items.pushMap(item);
+          }
+          promise.resolve(items);
+      });
+    });
```

Comment:
same here are we going to return something in case the user doesn't own any product?